src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1727,6 +1727,15 @@ format layout_optimizer::get_preferred_format(program_node& node) {
         if (use_onednn_impls) {
             expected = node.get_preferred_output_fmt();
         }
+
+        if (!allow_new_shape_infer && node.is_type<fully_connected>()) {
+            auto& fc_node = node.as<fully_connected>();
+            auto input_layout = fc_node.input().get_output_layout();
+            if (input_layout.format.dimension() > 4) {
```

Comment:
Do we need this condition? Seems that it is always setting it bfyx

Reply:
For static case, 4d input could be others like yxfb. In this case, some tests are failed if it forces to set to bfyx.